source/Company.py

```diff
@@ -1,53 +1,184 @@
+    '''
+    A function which checks if a given number is int
+    '''
+    if isinstance(val, int):
+        return True
+    return False
+
+
+def check_positive(num: float):
+    '''
+    A function which returns True if the given number
+    is positive and False otherwise
+    '''
+    if num <= 0:
+        return False
+    return True
+
 
 class Company:
     # you can add methods for your usage
 
-    def __init__(self, name, stocks_num, stock_price, comp_type):
-        pass
+    # The constructor function
+    def __init__(self, name: str, stocks_num: int,
+                 stock_price: float, comp_type: str):
+        if not check_string(name):
+            raise ValueError("Error! Company name starts with capital letter.")
+        else:
+            self.name = name
+        if check_positive(stocks_num) is False or check_int(stocks_num) is False:
+            raise ValueError("Error! Number of stocks must be positive.")
+        else:
+            self.stocks_num = stocks_num
+        if not check_positive(stock_price):
+            raise ValueError("Error! Price of stock must be positive.")
+        else:
+            self.stock_price = stock_price
+        if not check_string(comp_type):
+            raise ValueError("Error! Company type starts with capital letter.")
+        else:
+            self.comp_type = comp_type
 
     def net_worth(self):
-
-        pass
+        '''
+        Net worth of a company: as simple as that
+        '''
+        return self.stock_price*self.stocks_num
 
     def set_name(self, name):
-
-        pass
+        '''
+        Updates the company name, as long as its valid
+        '''
+        if name == self.name or check_string(name) is False:
+            return False
+        else:
+            self.name = name
+            return True
 
     def set_stocks_num(self, stocks_num):
-
-        pass
+        '''
+        Updates the number of stocks, as long as its valid
+        '''
+        if check_positive(stocks_num) is False or check_int(stocks_num) is False:
+            return False
+        else:
+            net_worth = self.net_worth()
+            self.stocks_num = stocks_num
+            self.stock_price = net_worth/self.stocks_num
+            return True
 
     def set_stock_price(self, stock_price):
-
-        pass
+        '''
+        Updates the price of a stock, as long as its valid
+        '''
+        if stock_price > self.net_worth() or check_positive(stock_price) is False:
+            return False
+        else:
+            net_worth = self.net_worth()
+            self.stock_price = stock_price
+            self.stocks_num = int(net_worth//self.stock_price)
+            return True
 
     def set_comp_type(self, comp_type):
-
-        pass
+        '''
+        Updates the type of the copmany, as long as its valid
+        '''
+        if not check_string(comp_type):
+            return False
+        else:
+            self.comp_type = comp_type
+            return True
```

Comment:
Why not check the Positive case, and return false at the end?
than you can delete the ```not``` in the if statement, and the else:
```python
if check_string(comp_type):
    self.comp_type = comp_type
    return True
return False
```

Reply:
Fixed! This is much better